### PR TITLE
Add the PassRateMetadata and FailuresMetadata models

### PR DESCRIPTION
**Pull request by lukebjerring**
Also adds a Golang based populate_dev_data.go script for adding some example output for the data.

### CODE REVIEW

File: metrics/naming.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	"strings"
+)
+
+// GetDatastoreKindName gets the full (namespaced) data type name for the given
+// interface (whether a pointer or not).
+func GetDatastoreKindName(data interface{}) string {
+	dataType := reflect.TypeOf(data)
+	for dataType.Kind() == reflect.Ptr {
+		dataType = reflect.Indirect(reflect.ValueOf(
+			data)).Type()
```
Comment (rwaldron) [inline metrics/naming.go]: Did `gofmt` do that? Is it a max column line break rule?

File: util/populate_dev_data.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+// populate_dev_data.go populates a local running webapp instance with some
+// of the latest production entities, so that there's data to view.
+//
+// It uses the AppEngine Remote API, which requires credentials; see:
+// https://cloud.google.com/appengine/docs/standard/go/tools/remoteapi/
+// https://developers.google.com/identity/protocols/application-default-credentials
+//
+// Usage (from util/):
+// go run populate_dev_data.go
```
Comment (rwaldron) [inline util/populate_dev_data.go]: I pulled this locally, restarted the web server and attempted to run: 

```
$ go run util/populate_dev_data.go
util/populate_dev_data.go:12:2: cannot find package "golang.org/x/oauth2/google" in any of:
	/usr/local/go/src/golang.org/x/oauth2/google (from $GOROOT)
	/Users/rwaldron/go/src/golang.org/x/oauth2/google (from $GOPATH)
```

File: metrics/models.go [hunk 3]
```diff
@@ -176,15 +176,24 @@ type TestRunStatus struct {
...
 
-type MetricsRun struct {
-	StartTime *time.Time    `json:"start_time"`
-	EndTime   *time.Time    `json:"end_time"`
-	TestRuns  *TestRunSlice `json:"test_runs"`
+type PassRateMetadata struct {
+	StartTime time.Time      `json:"start_time"`
+	EndTime   time.Time      `json:"end_time"`
+	TestRuns  []base.TestRun `json:"test_runs"`
+	DataUrl   string         `json:"url"`
```
Comment (rwaldron) [inline metrics/models.go]: Can we name this `\`json:"data_url"\`` to match the others?

File: metrics/models.go [hunk 4]
```diff
@@ -176,15 +176,24 @@ type TestRunStatus struct {
...
+}
+
+type FailuresMetadata struct {
+	StartTime   time.Time      `json:"start_time"`
+	EndTime     time.Time      `json:"end_time"`
+	TestRuns    []base.TestRun `json:"test_runs"`
+	DataUrl     string         `json:"url"`
```
Comment (rwaldron) [inline metrics/models.go]: Can we name this `\`json:"data_url"\`` to match the others?

File: metrics/naming.go (hunk 1 above)
Comment (lukebjerring) [reply metrics/naming.go]: We enforce gofmt with a lint-rule, so it's at a minimum gofmt validated. I can try reformat.

Comment (rwaldron) [reply metrics/naming.go]: > We enforce gofmt with a lint-rule

Yes, I know—I was just wondering if `gofmt` made this strange formatting choice. It's odd to read—maybe that's just me? Maybe it's not strange at all?

**Comment by lukebjerring** [comment]
Superseded by the broader https://github.com/w3c/wptdashboard/pull/334.

**Pull request closed without merging**